optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/api/Break.java

```diff
@@ -16,10 +16,34 @@
 
 package org.optaplanner.examples.common.experimental.api;
 
+/**
+ * A Break is a gap between two consecutive values. For instance,
+ * the list [1,2,4,5,6,10] has a break of length 2 between 2 and 4,
+ * as well as a break of length 4 between 6 and 10.
+ *
+ * @param <ValueType_> The type of value in the sequence
+ * @param <DifferenceType_> The type of difference between values in the sequence
+ */
 public interface Break<ValueType_, DifferenceType_ extends Comparable<DifferenceType_>> {
+    /**
+     * Return the item this break is directly after. For the
+     * break between 6 and 10, this will return 6.
+     * @return never null, the item this break is directly after
+     */
     ValueType_ getAfterItem();
 
+    /**
+     * Return the item this break is directly before. For the
+     * break between 6 and 10, this will return 10.
+     * @return never null,  the item this break is directly before
+     */
     ValueType_ getBeforeItem();
```

Comment:
Hmm, interesting. I was thinking about the API in the exact oposite way - getAfterItem() returns 10, because it is after the break. Maybe we need a better name for these two methods, one that is not ambiguous?

Reply:
`getStart()` and `getEnd()`?

Reply:
Well, it's technically not start or end. It's the thing before start, and thing after end. So the original naming was correct, but ambiguous.

getSequenceBeforeStart()/getSequenceAfterEnd() ?
(Note I replaced "item" with "sequence". It may be too specific, and we may need a more generic word, but I don't like that "item" is so generic that it doesn't say anything.)

Reply:
Then we would need to change the return type; I would expect `getSequenceBeforeStart()` be a sequence not an item. Start and End are still applicable (the last item of the sequence before is the start of the break, the first item of the sequence after is the end of the break).

Reply:
Good point! I did not notice that this actually referenced items, not sequences.
Yes, I think that this should point to preceding and succeeding sequence, not the actual item.
The items are accessible from the sequence. From the item, you can not get to the sequence.